font/src/ft/mod.rs

```diff
@@ -536,48 +583,48 @@ impl FreeTypeRasterizer {
         }
     }
 
-    fn load_face_with_glyph(&mut self, glyph: char) -> Result<FontKey, Error> {
-        let mut charset = fc::CharSet::new();
-        charset.add(glyph);
-        let mut pattern = fc::Pattern::new();
-        pattern.add_charset(&charset);
-        pattern.add_pixelsize(self.pixel_size as f64);
-
-        let config = fc::Config::get_current();
-        match fc::font_match(config, &mut pattern) {
-            Some(pattern) => {
-                if let (Some(path), Some(_)) = (pattern.file(0), pattern.index().nth(0)) {
-                    match self.keys.get(&path) {
-                        // We've previously loaded this font, so don't
-                        // load it again.
-                        Some(&key) => {
-                            debug!("Hit for font {:?}; no need to load", path);
-                            // Update fixup factor
-                            self.faces.get_mut(&key).unwrap().pixelsize_fixup_factor =
-                                pattern.pixelsizefixupfactor().next();
-                            Ok(key)
-                        },
-
-                        None => {
-                            debug!("Miss for font {:?}; loading now", path);
-                            // Safe to unwrap the option since we've already checked for the path
-                            // and index above.
-                            let key = self.face_from_pattern(&pattern)?.unwrap();
-                            Ok(key)
-                        },
+    fn load_face_with_glyph(&mut self, glyph: GlyphKey) -> Result<FontKey, Error> {
+        let ch = glyph.c;
+        let fallback_list = self.fallback_lists.get(&glyph.font_key).unwrap();
+        for font_pattern in fallback_list {
+            if let Some(path) = font_pattern.file(0) {
```

Comment:
Giant `if let` -> usually better to `match` and `continue` rather than permanently bumping indentation.